microsim/opencl/ramp/kernels/ramp_ua.cl

```diff
@@ -302,10 +316,33 @@ kernel void people_update_statuses(uint npeople,
         {
           // Calculate recovered prob based on age
           ushort person_age = people_ages[person_id];
-          float recovery_prob = get_recovery_prob_for_age(person_age, params);
+          float mortality_prob = get_mortality_prob_for_age(person_age, params);
+
+          ushort person_obesity = people_obesity[person_id]; 
+          if (person_obesity > 0){ // if person is obese then adjust mortality probability
+            mortality_prob *= get_obesity_multiplier(person_obesity, params);
+          }
+          
+          // if person has cardiovascular disease then adjust mortality probability
+          ushort person_cvd = people_cvd[person_id];
+          if (person_cvd){

```

Comment:
The compiler is _probably_ smart enough to optimise this, but I would typically write:
```c
mortality_prob *= (float)person_cvd * params->cvd_multiplier;
```
Just to ensure this doesn't branch (and then result in branch divergence).

Reply:
https://godbolt.org/z/szxfeo Okay at least one compiler isn't smart enough to avoid the branch. I would probably prefer the above form then.

Reply:
yep good call, this is nicer

Reply:
actually wait I just tried to implement this and I'm not sure you can do it without using a conditional, since person_cvd is zero if they don't have cardiovascular disease, but we don't want to multiply the mortality risk by zero